### PR TITLE
Block parser

### PR DESCRIPTION
**Pull request by Alfriadox**

### CODE REVIEW

File: wright/src/grammar/parsers/expression/block.rs [hunk 1]
```diff
@@ -1,12 +1,41 @@
...
+use crate::grammar::parsers::with_input;
+use nom::bytes::complete::tag;
+use nom::combinator::{map, opt};
+use nom::multi::many0;
+use nom::sequence::{delimited, pair, terminated};
 use nom::IResult;
 
 impl<'s> Block<'s> {
+    /// Block delimiters. Should not change.
+    pub const DELIMITERS: (&'static str, &'static str) = ("{", "}");
```
Comment (io12) [inline wright/src/grammar/parsers/expression/block.rs:14]: Consider making this a `[&'static str; 2]`

Comment (io12) [reply wright/src/grammar/parsers/expression/block.rs:14]: Actually `DELIMITER_LEFT` and `DELIMITER_RIGHT` might be better

File: wright/src/grammar/parsers/expression/block.rs [hunk 2]
```diff
@@ -1,12 +1,41 @@
...
+
```
Comment (io12) [inline wright/src/grammar/parsers/expression/block.rs:15]: Suggest adding a comment like
```rust
/// Parse the portion of a block inside the surrounding braces.
```

File: wright/src/grammar/parsers/expression/block.rs [hunk 3]
```diff
@@ -1,12 +1,41 @@
...
+    fn inner(
+        input: Fragment<'s>,
+    ) -> IResult<Fragment<'s>, (Vec<Statement<'s>>, Option<Box<Expression<'s>>>)> {
+        pair(
+            many0(terminated(Statement::parse, token_delimiter)),
+            opt(map(Expression::parse, |e| Box::new(e))),
+        )(input)
```
Comment (io12) [inline wright/src/grammar/parsers/expression/block.rs:22]: Is there any syntax that correctly parses as both a statement and expression? (A conditional expression maybe?) If so, the result expression will incorrectly be `None`.

Comment (Alfriadox) [reply wright/src/grammar/parsers/expression/block.rs:22]: There shouldn't be. Conditionals are always expressions

File: wright/src/grammar/parsers/expression/block.rs (hunk 1 above)
Comment (Alfriadox) [reply wright/src/grammar/parsers/expression/block.rs]: I'll refactor with `Start_delimiter` and `end_delimiter`

**Pull request merged**